lib/ast/function_dictionary.rb

```diff
@@ -0,0 +1,26 @@
+# a dictionary to hold function type definitions
+class FunctionDictionary
+  @dictionary = Hash.new { Hash.new }
+
+  def self.[](function_name)
+    @dictionary[function_name]
+  end
+
+  def self.add(name, type_signature)
+    @dictionary[name] = type_signature
+  end
+
+  @dictionary[:+] = { %i[INT INT] => :INT }
+  @dictionary[:-] = { %i[INT INT] => :INT }
+  @dictionary[:%] = { %i[INT INT] => :INT }
+  @dictionary[:/] = { %i[INT INT] => :INT }
+  @dictionary[:*] = { %i[INT INT] => :INT }
+  @dictionary[:"="] = { %i[INT INT] => :INT }
+  @dictionary[:<=] = { %i[INT INT] => :INT }
+  @dictionary[:<] = { %i[INT INT] => :INT }
+  @dictionary[:>=] = { %i[INT INT] => :INT }
+  @dictionary[:>] = { %i[INT INT] => :INT }
+  @dictionary[:!] = { %i[INT] => :INT }
+  @dictionary[:print] = { %i[STRING] => :INT }
+  @dictionary[:concat] = { %i[STRING STRING] => :STRING }
```

Comment:
Maybe move these out to another file and use the provided method for adding functions to the dictionary